### PR TITLE
refactor(commit): use custom conventional parser

### PR DESCRIPTION
**Pull request by bartsmykla**
## Motivation

Remove external dependency on `github.com/leodido/go-conventionalcommits` library which added complexity through:
- A fragile error-message-based fallback mechanism for trailer validation issues
- Dual parsing modes (library + manual fallback)
- Additional transitive dependency (`github.com/sirupsen/logrus`)

## Implementation information

- Add `parseTitle()` function with regex capture groups for conventional commit title parsing
- Remove `machine` field from `CommitParser` struct
- Remove `handleParseError()` fallback function and `extractLibraryBodyAndFooters()` helper
- Update `footerPattern` regex to support "BREAKING CHANGE" (with space per conventional commits spec)
- Add unit tests for `parseTitle()` covering valid titles, invalid titles, and edge cases

> Changelog: skip

### CODE REVIEW

File: internal/validators/git/commit_parser.go [hunk 1]
```diff
@@ -3,14 +3,41 @@ package git
...
-
-	conventionalcommits "github.com/leodido/go-conventionalcommits"
-	ccp "github.com/leodido/go-conventionalcommits/parser"
 )
 
 // footerPattern matches git trailer format: "Token: value"
+// Supports "BREAKING CHANGE" (with space) per conventional commits spec.
 // Compiled once at package initialization for efficiency.
-var footerPattern = regexp.MustCompile(`^([A-Za-z0-9-]+):\s*(.*)$`)
+var footerPattern = regexp.MustCompile(`^([A-Za-z0-9][A-Za-z0-9- ]*):\s*(.*)$`)
```
[the remark this answers is not in the document]
Comment (bartsmykla) [reply internal/validators/git/commit_parser.go]: addressed

File: internal/validators/git/commit_parser.go [hunk 2]
```diff
@@ -108,40 +131,30 @@ func (p *CommitParser) Parse(message string) *ParsedCommit {
...
+		result.IsBreakingChange = true
 	}
 
-	// Only extract body/footers from the library if we didn't use fallback mode.
-	// In fallback mode, we already manually extracted these via extractBodyAndFooters().
-	if !usedFallback {
-		p.extractLibraryBodyAndFooters(cc, result)
+	if _, hasBreaking := result.Footers["BREAKING-CHANGE"]; hasBreaking {
+		result.IsBreakingChange = true
 	}
```
[the remark this answers is not in the document]
Comment (bartsmykla) [reply internal/validators/git/commit_parser.go]: Fixed. Removed the redundant BREAKING CHANGE detection from `Parse()` since `extractFootersFromLines()` already handles this. Updated the comment to indicate the function also sets `IsBreakingChange`.

**Comment by bartsmykla** [comment]
squash
